### PR TITLE
Version shared objects

### PR DESCRIPTION
**Pull request by sanjayankur31**
First attempt at fixing #624 

With this, the libraries now look like this:

```
$ ls
libclipper.so
libclipper.so.2.0
libclipper.so.2.0.2
libnetlib.so
libnetlib.so.2.0
libnetlib.so.2.0.2
libopenjpeg2.so
libopenjpeg2.so.2.0
libopenjpeg2.so.2.0.0
librply.so
librply.so.2.0
librply.so.2.0.2
libs.txt
libtestlib.so
libtestlib.so.2.0
libtestlib.so.2.0.2
libv3p_netlib.so
libv3p_netlib.so.2.0
libv3p_netlib.so.2.0.2
libvbl_io.so
libvbl_io.so.2.0
libvbl_io.so.2.0.2
libvbl.so
libvbl.so.2.0
libvbl.so.2.0.2
libvcl.so
libvcl.so.2.0
libvcl.so.2.0.2
libvcsl.so
libvcsl.so.2.0
libvcsl.so.2.0.2
libvgl_algo.so
libvgl_algo.so.2.0
libvgl_algo.so.2.0.2
libvgl_io.so
libvgl_io.so.2.0
libvgl_io.so.2.0.2
libvgl.so
libvgl.so.2.0
libvgl.so.2.0.2
libvgl_xio.so
libvgl_xio.so.2.0
libvgl_xio.so.2.0.2
libvil1.so
libvil1.so.2.0
libvil1.so.2.0.2
libvil_algo.so
libvil_algo.so.2.0
libvil_algo.so.2.0.2
libvil_io.so
libvil_io.so.2.0
libvil_io.so.2.0.2
libvil.so
libvil.so.2.0
libvil.so.2.0.2
libvnl_algo.so
libvnl_algo.so.2.0
libvnl_algo.so.2.0.2
libvnl_io.so
libvnl_io.so.2.0
libvnl_io.so.2.0.2
libvnl.so
libvnl.so.2.0
libvnl.so.2.0.2
libvnl_xio.so
libvnl_xio.so.2.0
libvnl_xio.so.2.0.2
libvpdl.so
libvpdl.so.2.0
libvpdl.so.2.0.2
libvpgl_algo.so
libvpgl_algo.so.2.0
libvpgl_algo.so.2.0.2
libvpgl_file_formats.so
libvpgl_file_formats.so.2.0
libvpgl_file_formats.so.2.0.2
libvpgl_io.so
libvpgl_io.so.2.0
libvpgl_io.so.2.0.2
libvpgl.so
libvpgl.so.2.0
libvpgl.so.2.0.2
libvpgl_xio.so
libvpgl_xio.so.2.0
libvpgl_xio.so.2.0.2
libvpl.so
libvpl.so.2.0
libvpl.so.2.0.2
libvsl.so
libvsl.so.2.0
libvsl.so.2.0.2
libvul_io.so
libvul_io.so.2.0
libvul_io.so.2.0.2
libvul.so
libvul.so.2.0
libvul.so.2.0.2

$ for f in *.so*; do readelf -a $f | grep SONAME; done 
 0x000000000000000e (SONAME)             Library soname: [libclipper.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libclipper.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libclipper.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libnetlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libnetlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libnetlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libopenjpeg2.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libopenjpeg2.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libopenjpeg2.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [librply.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [librply.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [librply.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libtestlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libtestlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libtestlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libv3p_netlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libv3p_netlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libv3p_netlib.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvbl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvbl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvbl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvbl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvbl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvbl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvcl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvcl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvcl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvcsl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvcsl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvcsl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvgl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil1.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil1.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil1.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvil.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvnl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpdl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpdl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpdl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_algo.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_file_formats.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_file_formats.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_file_formats.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpgl_xio.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvpl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvsl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvsl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvsl.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvul_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvul_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvul_io.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvul.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvul.so.2.0]
 0x000000000000000e (SONAME)             Library soname: [libvul.so.2.0]
```

### CODE REVIEW

**Comment by decrispell** [comment]
makes sense to me, but this type of thing is a little out of my comfort zone - Any comments @andyneff @mleotta @hjmjohnson  ?

**Comment by andyneff** [comment]
I'm not familiar enough with cmake to say anything more then "I agree this should be done"

@decrispell Should the `set_target_properties` be added to the `vxl_add_library` macro? (So that you don't have to remember to add the `set_target_properties` every time)

**Comment by sanjayankur31** [comment]
> I'm not familiar enough with cmake to say anything more then "I agree this should be done"
> 
> @decrispell Should the `set_target_properties` be added to the `vxl_add_library` macro? (So that you don't have to remember to add the `set_target_properties` every time)

I could look into that if that'll make it easier + clearer.

This PR goes with a few other changes too:

- use of semantic versioning implies the release process follows the guide here: https://semver.org
- the nightly bot should be disabled (and the version information no longer needs to be hard coded in the source file---it can be picked from the cmake variables if that is required)

**Comment by sanjayankur31** [comment]
The `vxl_add_library` macro already used `set_target_properties` so there was no need to define library versions for each library. Setting the `SOVERSION` should be enough. I updated the PR accordingly.

**Comment by sanjayankur31** [comment]
Hello! What do we think about this? Can we proceed to review + merge (and turn of the autoversion bot too)?

File: CMakeLists.txt [hunk 1]
```diff
@@ -67,10 +67,17 @@ foreach(p
 endforeach()
 
 project(VXL #Project name must be all caps to have properly generated VXL_VERSION_* variables
-    VERSION 2.0.2.0
```
Comment (hjmjohnson) [inline CMakeLists.txt]: Please do not remove this line.  With CMake 3.10.2 this line defines

The project() command stores the version number and its components in variables
```
PROJECT_VERSION, <PROJECT-NAME>_VERSION
PROJECT_VERSION_MAJOR, <PROJECT-NAME>_VERSION_MAJOR
PROJECT_VERSION_MINOR, <PROJECT-NAME>_VERSION_MINOR
PROJECT_VERSION_PATCH, <PROJECT-NAME>_VERSION_PATCH
PROJECT_VERSION_TWEAK, <PROJECT-NAME>_VERSION_TWEAK
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -67,10 +67,17 @@ foreach(p
...
     DESCRIPTION "A multi-platform collection of C++ software libraries for Computer Vision and Image Understanding."
     LANGUAGES CXX C)
 
+SET(VXL_MAJOR_VERSION 2)
```
Comment (hjmjohnson) [inline CMakeLists.txt]: Use VXL_VERSION_MAJOR instead as set in the project command.

**Comment by sanjayankur31** [comment]
I think given that vxl is quite an important library that many other tools depend on, announcing API/ABI changes using semver would be a good idea.

If any other changes are need to this PR, please do let me know. 

Cheers!

**Pull request merged**